### ISSUE TITLE
What's the reasoning behind createUploadedFile renaming and moving to temporary path?

### ISSUE THREAD

**Issue by justinlevi** [symfony/psr-http-message-bridge#39]
@Danielss89 
In HTTPFoundationFactory we have the following code: 

```
        if (UPLOAD_ERR_NO_FILE !== $psrUploadedFile->getError()) {
            $temporaryPath = $this->getTemporaryPath();
            $psrUploadedFile->moveTo($temporaryPath);
            $clientFileName = $psrUploadedFile->getClientFilename();
        }
```

https://github.com/symfony/psr-http-message-bridge/blob/master/Factory/HttpFoundationFactory.php#L102

What's the rationale of moving to a new temporary path here vs keeping the same file name? 

I ask because this ends up causing an issue if a single POST request is split, say in the case of an authentication check subrequest, if the the original request includes a $_Files object.  Would it be bad to just leave the temporary files where they are with the same names for some reason? Just looking for some background on that decision.

**Comment by Nyholm** [comment]
I would assume it is because you should not have two abstractions over the same file. But Im not sure.

**Comment by nicolas-grekas** [comment]
Most other issues tell about upload files, maybe something where someone should have a look.
PR welcome.

**Comment by BunnyHolder** [comment]
This is pain in my ass now. Worst part is crazy path where it tries to move file `tempnam(sys_get_temp_dir(), uniqid('symfony', true));`

**Comment by nicolas-grekas** [comment]
Fixed in #72 / v1.3.0
